src/Workspaces/Core/Portable/Workspace/Solution/TextDocument.cs

```diff
@@ -17,11 +17,7 @@ public class TextDocument
         /// <summary>
         /// The project this document belongs to.
         /// </summary>
-        public Project Project { get; protected set; }
-
-        protected TextDocument()
```

Comment:
💡 

```csharp
[Obsolete("This API does not work and should not be used", error: true)]
[EditorBrowsable(EditorBrowsableState.Never)]
protected TextDocument() => throw new NotSupportedException();
```

Reply:
Not sure how that's better; it'll still break build and still break runtime...

Reply:
You could change the body to

```
{
  // This should not be called since it leaves the type in an invalid state, but we don't want to break the shipped public API
  State = null!;
}
```

Reply:
There's still no way I can imagine somebody is going to use that. I'm just going to pull the band-aid.